File: Assets/Scripts/BotonPresionado.js
#pragma strict

var interval:float = 0.25f;

var mIsPressed:boolean = false;
var mNextClick:float = 0f;
private var motor : CharacterMotorNemoris;

// Use this for initialization
function Awake () {
	motor = GameObject.FindWithTag("Player").GetComponent(CharacterMotorNemoris);
}

function OnPress (isPressed:boolean) { 
	mIsPressed = isPressed; 
	mNextClick = Time.realtimeSinceStartup + interval; 
}
    
function Update (){
	motor.inputJump = false;
	if(Input.GetKeyDown(KeyCode.Space)){
		mIsPressed = true; 
		mNextClick = Time.realtimeSinceStartup + interval; 
		print("jump");
	}
	if(Input.GetKeyUp(KeyCode.Space)){
		mIsPressed = false; 
		print("jump2");
	}
	
	if (mIsPressed && Time.realtimeSinceStartup < mNextClick)
	{
	    mNextClick = Time.realtimeSinceStartup + interval;

	    // Do what you need to do, or simply:
	    //SendMessage("OnClick", SendMessageOptions.DontRequireReceiver);
	    motor.inputJump = true;
	}
	
}

function Start () {

}

File: Assets/Scripts/new/BotonPresionado.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class BotonPresionado : MonoBehaviour
{
    public float interval = 0.25f;

    public bool mIsPRessed = false;
    public float mNextClick = 0;
    private CharController motor;

    void Awake(){
        motor = GameObject.FindGameObjectWithTag("Player").GetComponent<CharController>();
    }

    void OnPress(bool isPressed){
        mIsPRessed = isPressed;
        mNextClick = Time.realtimeSinceStartup + interval;
    }

    void Update(){
        motor.inputJump = false;
        if(Input.GetKeyDown(KeyCode.Space)){
            mIsPRessed = true;
            mNextClick = Time.realtimeSinceStartup + interval;
            print("jump");
        }
        if(Input.GetKeyUp(KeyCode.Space)){
            mIsPRessed = false;
            print("jump2");
        }

        if(mIsPRessed && Time.realtimeSinceStartup < mNextClick){
            mNextClick = Time.realtimeSinceStartup + interval;
            motor.inputJump = true;
        }
    }
}
